programmers/로또의-최고-순위와-최저-순위.js

```diff
@@ -0,0 +1,13 @@
+function solution(lottos, win_nums) {
+    const notSameNum = lottos.filter(x => !win_nums.includes(x));
+    const sameNum = lottos.filter(x => win_nums.includes(x));
+    const zeroCnt = lottos.filter((num) => num === 0).length
+    
+    if(zeroCnt === 6){
+        return [1, 6];
+    } else if(sameNum.length === 0) {
+        return [6, 6];
+    }
+    const result = [7 - (sameNum.length + zeroCnt), 7 - sameNum.length];
+    return result;
+}
```

Comment:
```suggestion
function solution(lottos, win_nums) {
    const sameNum = lottos.filter(x => win_nums.includes(x));
    const zeroCnt = lottos.filter((num) => !num).length;

    if (zeroCnt === 6){
        return [1, 6];
    } else if (!sameNum.length) {
        return [6, 6];
    }
    
    return [7 - (sameNum.length + zeroCnt), 7 - sameNum.length];
}
```
1. filter 👍
2. notSameNum은 없어도 잘 돌아가네요 !
3. 기본적으로 문장끝에 세미콜론 필수
4. result 변수 없이 바로 return 가능 

코드를 작성하면서 컨벤션을 맞추면 제일 좋지만, 어느정도의 코드를 짜면 
컨벤션을 한번씩 점검 해주면 좋을것같고, 변수명은 내가 알아보는건 물론이고
남들한테 보여줬을때 어떤 역할을 하는 변수인지 알 수 있게끔 만들면 좋아요

```js
function solution(lottos, win_nums) {
    const ranks = [6, 6, 5, 4, 3, 2, 1];
    const zeroCnt = lottos.filter((num) => !num).length;
    const winningCnt = lottos.filter((win) => win_nums.includes(win)).length;
    
    return [ranks[winningCnt + zeroCnt],ranks[winningCnt]]
}
```